kitsune/search/v2/search.py

```diff
@@ -0,0 +1,238 @@
+    return None
+
+
+def sanitize(summary):
+    return bleach.clean(
+        summary,
+        tags=[HIGHLIGHT_TAG],
+        strip=True,
+    )
+
+
+class BaseConfig(ABC):
+    """Semi-abstract base class for Search configs."""
+
+    @property
+    @abstractmethod
+    def index(self):
+        pass
+
+    @property
+    @abstractmethod
+    def fields(self):
+        """An array of fields to search over."""
+        pass
+
+    @property
+    @abstractmethod
+    def highlight_fields(self):
+        """An array of fields to highlight."""
+        pass
+
+    @property
+    def filter(self):
+        """A filter which returns all documents of this type to search over."""
+        return Q("term", _index=self.index)
+
+    @abstractmethod
+    def product_filter(self, product):
+        """A filter which returns documents with the correct product."""
+        pass
+
+    @abstractmethod
+    def make_result(self, hit, locale):
+        """Transform a hit into a result dictionary."""
+        pass
+
+
+class AAQConfig(BaseConfig):
+    """Search config for AAQ."""
+
+    index = QUESTION_INDEX_NAME
+    fields = [
+        "question_title.{}^4",
+        "question_content.{}^3",
+        "answer_content.{}^3",
+    ]
+    highlight_fields = ["question_content.{}", "answer_content.{}"]
+
+    @property
+    def filter(self):
+        return Q(
+            "bool",
+            filter=super().filter,
+            # exclude AnswerDocuments from the search:
+            must_not=Q("exists", field="updated"),
+        )
+
+    def product_filter(self, product):
+        return Q("term", question_product_id=product.id)
+
+    def make_result(self, hit, locale):
+        summary = first_highlight(hit)
+        if not summary:
+            summary = hit.question_content[locale][:500]
+        summary = sanitize(summary)
+
+        num_answers = 0
+        answer_content = getattr(hit, "answer_content", None)
+        if answer_content:
+            num_answers = len(answer_content[locale])
+
+        return {
+            "type": "question",
+            "url": "/questions/{}".format(hit.question_id),
+            "score": hit.meta.score,
+            "title": hit.question_title[locale],
+            "search_summary": summary,
+            "last_updated": datetime.fromisoformat(hit.question_updated),
+            "is_solved": hit.question_has_solution,
+            "num_answers": num_answers,
+            "num_votes": hit.question_num_votes,
+        }
+
+
+class KBConfig(BaseConfig):
+    """Search config for the Knowledge Base."""
+
+    index = WIKI_DOCUMENT_INDEX_NAME
+    fields = [
+        "keywords.{}^8",
+        "title.{}^6",
+        "summary.{}^2",
+        "content.{}^1",
+    ]
+    highlight_fields = ["summary.{}", "content.{}"]
+
+    @property
+    def filter(self):
+        return Q(
+            "bool",
+            filter=super().filter,
+            must_not=Q("terms", category=[TEMPLATES_CATEGORY, CANNED_RESPONSES_CATEGORY]),
```

Comment:
do we need this at all? Currently I am not sure if these categories are used at all

Reply:
Yeah, results from these categories were popping up locally without it - perhaps we should filter by the categories we do want to search over, rather than those which we don't